src/main/java/pl/edu/pw/mini/ingreedio/api/controller/ProductController.java

```diff
@@ -85,6 +87,35 @@ public ResponseEntity<Product> addProduct(@RequestBody Product product) {
         return new ResponseEntity<>(savedProduct, HttpStatus.CREATED);
     }
 
+    @Operation(summary = "Delete product from the database",
+        description = "Delete product from the database",
+        security = {@SecurityRequirement(name = "Bearer Authentication")})
+    @PreAuthorize("hasAuthority('REMOVE_PRODUCT')")
+    @DeleteMapping
+    @ResponseBody
+    public ResponseEntity<Void> deleteProduct(@PathVariable Long id) {
+        boolean deleted = productService.deleteProduct(id);
+        if (deleted) {
+            return ResponseEntity.ok().build();
+        }
+        return ResponseEntity.notFound().build();
+    }
+
+    @Operation(summary = "Edit product in the database",
+        description = "Edit product in the database",
+        security = {@SecurityRequirement(name = "Bearer Authentication")})
+    @PreAuthorize("hasAuthority('EDIT_PRODUCT')")
+    @PutMapping("/{id}")
+    @ResponseBody
+    public ResponseEntity<Product> editProduct(@PathVariable Long id,
+                                               @RequestBody Product product) {
+        Optional<Product> editedProduct = productService.editProduct(id, product);
+        if (editedProduct.isPresent()) {
+            return ResponseEntity.ok().build();
+        }
+        return ResponseEntity.notFound().build();
+    }
```

Comment:
I think it would be useful to also have a PATCH mapping which changes only some of the data of product

Reply:
I thought that if we only want to change some of the data we can just send previous data in the request (because we have current fields) but I can change it

Reply:
We will discuss that later